From f8759e3c3b99e15d4b182e3240d0990590e6cc36 Mon Sep 17 00:00:00 2001
From: Mike Wester <mwester@dis.net>
Date: Sun, 13 Apr 2008 07:25:57 +0100
Subject: [PATCH] fix-gta01-spi-resume-patch

Patch to register the SPI device and thus the save/resume for the display

This patch does *NOT* resolve this problem, but it does make things a bit
better.  The SPI device name changed with 2.6.24, and the change was made for
the gta02 but never propagated to the gta01.  With this change, you'll now see
the entries for the spi bus and the display controller in /sys; and the display
controller suspend/resume functions will be called.

(via BZ 79)

Signed-off-by: Mike Wester <mwester@dis.net>
---
 arch/arm/mach-s3c2410/mach-gta01.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-s3c2410/mach-gta01.c b/arch/arm/mach-s3c2410/mach-gta01.c
index 8c1a402..a948311 100644
--- a/arch/arm/mach-s3c2410/mach-gta01.c
+++ b/arch/arm/mach-s3c2410/mach-gta01.c
@@ -529,6 +529,7 @@ static struct s3c2410_spigpio_info spi_gpio_cfg = {
 	.board_size	= ARRAY_SIZE(gta01_spi_board_info),
 	.board_info	= gta01_spi_board_info,
 	.chip_select	= &spi_gpio_cs,
+	.num_chipselect = 2,   /*** Should be 1 or 2 for gta01? ***/
 };
 
 static struct resource s3c_spi_lcm_resource[] = {
@@ -551,7 +552,7 @@ static struct resource s3c_spi_lcm_resource[] = {
 };
 
 struct platform_device s3c_device_spi_lcm = {
-	.name		  = "s3c24xx-spi-gpio",
+	.name		  = "spi_s3c24xx_gpio",
 	.id		  = 1,
 	.num_resources	  = ARRAY_SIZE(s3c_spi_lcm_resource),
 	.resource	  = s3c_spi_lcm_resource,
-- 
1.5.6.5

